python-package/xgboost/dask.py

```diff
@@ -839,6 +841,11 @@ def dispatched_train(
                     LOGGER.info(msg)
                 else:
                     local_param[p] = worker.nthreads
+            
+            # If worker did not receive input data, return without failing
+            if local_dtrain.num_row() == 0:
```

Comment:
You should let the train continue, otherwise the cluster can hang on waiting for this exited worker for synchronization.

Reply:
If I let training continue, I get this error `Check failed: mparam_.num_feature != 0 (0 vs. 0) : 0 feature is supplied. Are you using raw Booster interface?` in `learner.cc` if the worker did not receive input data.

Reply:
@elaineejiang  Thanks for the explanation.  In that case, the synchronization between DMatrix is failing.  During the construction, DMatrix on workers synchronize the number of columns in input data, see https://github.com/dmlc/xgboost/blob/72892cc80d2b656d0d911381ee29a32904308fb4/src/data/simple_dmatrix.cc#L159 .  So as long as one of the worker has any data, it shouldn't be 0.

Reply:
I see, thanks @trivialfis. Just to be clear, I'm only seeing this error when I try to run multiple XGBoost learners in parallel via
```
dask_client.submit(xgb.dask.DaskXGBRegressor([...]).fit(dd, xcols, ycol, wcol)
```
Is this something that's not recommend or not supported?

Reply:
I think the correct version should be something similar to:

``` python
classifier_future = client.submit(classifier.fit, X, y, sample_weight=w, eval_set=[(X, y)])
```

This should work.

Reply:
Ah so sorry -- I typed that out on the fly and got the syntax wrong. I am seeing
```
Check failed: mparam_.num_feature != 0 (0 vs. 0) : 0 feature is supplied. Are you using raw Booster interface?
```
with what your version.

If that's _supposed_ to work, I guess it's just my network issue?